cli/command/image/tree.go

```diff
@@ -0,0 +1,296 @@
+			continue
+		}
+		d := h.Width + headerSpacing
+		// If the first column gets too short, remove remaining columns
+		if nameWidth-d < 12 {
+			headers = headers[:idx]
+			break
+		}
+		nameWidth -= d
+	}
+
+	maxImageName := len(headers[0].Title)
+	for _, img := range images {
+		for _, name := range img.Names {
+			if len(name) > maxImageName {
+				maxImageName = len(name)
+			}
+		}
+		for _, sub := range img.Children {
+			if len(sub.Platform) > maxImageName {
+				maxImageName = len(sub.Platform)
+			}
+		}
+	}
+
+	if nameWidth > maxImageName+headerSpacing {
+		nameWidth = maxImageName + headerSpacing
+	}
+
+	if nameWidth < 0 {
+		headers = headers[:1]
+		nameWidth = int(width)
+	}
+	headers[0].Width = nameWidth
+
+	headerColor := aec.NewBuilder(aec.DefaultF, aec.Bold).ANSI
+	topNameColor := aec.NewBuilder(aec.BlueF, aec.Underline, aec.Bold).ANSI
+	normalColor := aec.NewBuilder(aec.DefaultF).ANSI
+	normalFaintedColor := aec.NewBuilder(aec.DefaultF).Faint().ANSI
+	greenColor := aec.NewBuilder(aec.GreenF).ANSI
+	if !out.IsTerminal() {
+		headerColor = noColor{}
+		topNameColor = noColor{}
+		normalColor = noColor{}
+		normalFaintedColor = noColor{}
+		greenColor = noColor{}
+	}
+
+	// Print headers
+	for i, h := range headers {
+		if i > 0 {
+			_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+		}
+
+		_, _ = fmt.Fprint(out, h.PrintC(headerColor, h.Title))
+	}
+
+	_, _ = fmt.Fprintln(out)
+
+	printDetails := func(clr aec.ANSI, details imageDetails) {
+		if len(headers) <= 1 {
+			return
+		}
+		truncID := stringid.TruncateID(details.ID)
+		_, _ = fmt.Fprint(out, headers[1].Print(clr, truncID))
+		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+
+		if len(headers) <= 2 {
+			return
+		}
+		_, _ = fmt.Fprint(out, headers[2].Print(clr, details.DiskUsage))
+		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+
+		if len(headers) <= 3 {
+			return
+		}
+		if details.Used {
+			_, _ = fmt.Fprint(out, headers[3].Print(greenColor, "✔"))
+		} else {
+			_, _ = fmt.Fprint(out, headers[3].Print(clr, " "))
+		}
+		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+	}
+
+	// Print images
+	for _, img := range images {
+		_, _ = fmt.Fprintln(out, "")
+		for nameIdx, name := range img.Names {
+			if nameIdx != 0 {
+				_, _ = fmt.Fprintln(out, "")
+			}
+			_, _ = fmt.Fprint(out, headers[0].Print(topNameColor, name))
+		}
+		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+
+		printDetails(normalColor, img.Details)
+
+		_, _ = fmt.Fprintln(out, "")
+		for idx, sub := range img.Children {
+			clr := normalColor
+			if !sub.Available {
+				clr = normalFaintedColor
+			}
+
+			if idx != len(img.Children)-1 {
+				_, _ = fmt.Fprint(out, headers[0].Print(clr, "├─ "+sub.Platform))
+			} else {
+				_, _ = fmt.Fprint(out, headers[0].Print(clr, "└─ "+sub.Platform))
+			}
+
+			_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
+			printDetails(clr, sub.Details)
+
+			_, _ = fmt.Fprintln(out, "")
+		}
+	}
```

Comment:
If we refactored a bit, this could become something line
```go
	for iid, img := range images {
		printImageNames(out, img, headers, topNameColor)
		printDetails(out, normalColor, img.Details, headers)
		printImageChildren(out, img, headers)
	}
```
and then we'd have
```go
func printImageNames(out io.Writer, img topImage, headers []header, topNameColor aec.ANSI) {
	for nameIdx, name := range img.Names {
		if nameIdx != 0 {
			_, _ = fmt.Fprintln(out, "")
		}
		_, _ = fmt.Fprint(out, headers[0].Print(topNameColor, name))
	}
	_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
}

func printDetails(out io.Writer, clr aec.ANSI, details imageDetails, headers []header) {
	if len(headers) <= 1 {
		return
	}
	truncID := stringid.TruncateID(details.ID)
	_, _ = fmt.Fprint(out, headers[1].Print(clr, truncID))
	_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))

	if len(headers) <= 2 {
		return
	}
	_, _ = fmt.Fprint(out, headers[2].Print(clr, details.DiskUsage))
	_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))

	if len(headers) <= 3 {
		return
	}
	if details.Used {
		_, _ = fmt.Fprint(out, headers[3].Print(greenColor, "✔"))
	} else {
		_, _ = fmt.Fprint(out, headers[3].Print(clr, " "))
	}
	_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
}

func printImageChildren(out io.Writer, img topImage, headers []header) {
	_, _ = fmt.Fprintln(out, "")
	for idx, sub := range img.Children {
		clr := normalColor
		if !sub.Available {
			clr = normalFaintedColor
		}

		if idx != len(img.Children)-1 {
			_, _ = fmt.Fprint(out, headers[0].Print(clr, "├─ "+sub.Platform))
		} else {
			_, _ = fmt.Fprint(out, headers[0].Print(clr, "└─ "+sub.Platform))
		}

		_, _ = fmt.Fprint(out, strings.Repeat(" ", headerSpacing))
		printDetails(out, clr, sub.Details, headers)

		_, _ = fmt.Fprintln(out, "")
	}
}
```

Reply:
Looks nicer, thanks!